modules/liveIntentAnalyticsAdapter.js

```diff
@@ -0,0 +1,147 @@
+import {ajax} from '../src/ajax.js';
+import { generateUUID, logInfo, logWarn } from '../src/utils.js';
+import adapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import CONSTANTS from '../src/constants.json';
+import adapterManager from '../src/adapterManager.js';
+import { auctionManager } from '../src/auctionManager.js';
+
+const ANALYTICS_TYPE = 'endpoint';
+const URL = 'https://wba.liadm.com/analytic-events';
+const GVL_ID = 148;
+const ADAPTER_CODE = 'liveintent';
+const DEFAULT_SAMPLING = 0.1;
+const DEFAULT_BID_WON_TIMEOUT = 2000;
+const { EVENTS: { AUCTION_END } } = CONSTANTS;
+let initOptions = {};
+let isSampled;
+let bidWonTimeout;
+
+function handleAuctionEnd(args) {
+  setTimeout(() => {
+    const auction = auctionManager.index.getAuction(args.auctionId);
+    const winningBids = (auction) ? auction.getWinningBids() : [];
+    const data = createAnalyticsEvent(args, winningBids);
+    sendAnalyticsEvent(data);
+  }, bidWonTimeout);
+}
+
+function getAnalyticsEventBids(bidsReceived) {
+  return bidsReceived.map(bid => {
+    return {
+      adUnitCode: bid.adUnitCode,
+      timeToRespond: bid.timeToRespond,
+      cpm: bid.cpm,
+      currency: bid.currency,
+      ttl: bid.ttl,
+      bidder: bid.bidder
+    };
+  });
+}
+
+function getBannerSizes(banner) {
+  if (banner && banner.sizes) {
+    return banner.sizes.map(size => {
+      const [width, height] = size;
+      return {w: width, h: height};
+    });
+  } else return [];
+}
+
+function getUniqueBy(arr, key) {
+  return [...new Map(arr.map(item => [item[key], item])).values()]
+}
+
+function createAnalyticsEvent(args, winningBids) {
+  let payload = {
+    instanceId: generateUUID(),
+    url: window.location.href,
```

Comment:
My proposal here is to use the method `getRefererInfo()` from file `refererDetection.js` that is robust and works in complex scenarios too.